web/components/charts/contract/binary.tsx

```diff
@@ -16,28 +15,29 @@ import {
 } from '../helpers'
 import { HistoryPoint, SingleValueHistoryChart } from '../generic-charts'
 import { Row } from 'web/components/layout/row'
-import { Avatar } from 'web/components/widgets/avatar'
+import { BetPoint } from 'web/pages/[username]/[contractSlug]'
 
 const MARGIN = { top: 20, right: 40, bottom: 20, left: 10 }
 const MARGIN_X = MARGIN.left + MARGIN.right
 const MARGIN_Y = MARGIN.top + MARGIN.bottom
 
-const getBetPoints = (bets: Bet[]) => {
-  return sortBy(bets, (b) => b.createdTime).map((b) => ({
-    x: new Date(b.createdTime),
-    y: b.probAfter,
+const getBetPoints = (bets: BetPoint[]) => {
+  return sortBy(bets, (b) => b.x).map((b) => ({
+    x: new Date(b.x),
+    y: b.y,
     obj: b,
   }))
 }
 
-const BinaryChartTooltip = (props: TooltipProps<Date, HistoryPoint<Bet>>) => {
+const BinaryChartTooltip = (
+  props: TooltipProps<Date, HistoryPoint<BetPoint>>
+) => {
   const { prev, x, xScale } = props
   const [start, end] = xScale.domain()
   const d = xScale.invert(x)
   if (!prev) return null
   return (
     <Row className="items-center gap-2">
-      {prev.obj && <Avatar size="xs" avatarUrl={prev.obj.userAvatarUrl} />}
```

Comment:
I'll miss the avatar. Seems like we should be able to keep it for small markets?

Could include the avatar url as optional in BetPoint and set it if there are e.g. less than 500 bets.

Reply:
I think it's probably worth keeping in. Something I mentioned to Ian is that another Actually Good Change for the whole website would be to make avatar URLs a function of the user ID, which would be very easy, and then we wouldn't have to be storing and sending bulky avatar URLs around everywhere.

Of course, with 10k bets, then a 20 character user ID with every bet is still 200k characters coming over the wire. So at some point we are just going to have to redesign the charts to not download every data point from forever by default.

Reply:
I'm now adding the avatar for contracts with less than 1000 bets. I could optionally include some avatars in the most recent 1000 bet points 🤔